### ISSUE TITLE
Unnecessary reconcile after deletion

### ISSUE THREAD

**Issue by muvaf** [crossplane/terrajet#109]
### What happened?

In debug logs, I see that for some reason the finalizer removal call doesn't succeed due to infamous `custom resource version is old` error and we get re-queued. Since we remove the directory before the finalizer call, in the next reconciliation we need to `init` the directory again to see whether resource exists and only after that we confirm the deletion. This causes the deletion to take longer than it needs to.

### How can we reproduce it?

Use any sync examples like [AWS VPC](https://github.com/crossplane-contrib/provider-tf-aws/blob/main/examples/ec2/vpc.yaml) with debug logging enabled.

**Comment by ulucinar** [comment]
I have tried to reproduce the issue in another attempt by creating 10 VPCs and deleting them using a modified Terrajet version which logs the deletion timestamp of the managed resources while initializing a Terraform workspace. I did not encounter any workspace initialization after destruction issue for these 10 resources. 

Closing this issue for the moment. If we encounter similar issues again, let's reopen it with the corresponding provider logs and the client-go version used.
